### PR TITLE
fix stack depot

### PR DESCRIPTION
**Pull request by dvyukov**
Build linked lists instead of always overwriting head.
Invert stack comparison predicate.
Don't calculate hash twice on fast path.
Remove unnecessary indirections in source.

### CODE REVIEW

File: mm/ktsan/stack_depot.c [hunk 1]
```diff
@@ -3,40 +3,29 @@
...
 		depot->parts[i] = NULL;
-	}
 	kt_cache_init(&depot->stack_cache, KT_STACK_DEPOT_MEMORY_LIMIT, 1);
 	depot->stack_offset = 0;
 	kt_spin_init(&depot->lock);
 }
 
 /* Only available during early boot. */
-void __init kt_stack_depot_destroy(kt_stack_depot_t *depot)
+void kt_stack_depot_destroy(kt_stack_depot_t *depot)
```
Comment (xairy) [inline mm/ktsan/stack_depot.c]: Why did you remove __init?

Comment (dvyukov) [inline mm/ktsan/stack_depot.c]: What is the intended usage scenario of this call during kernel loading?
I don't see any. If we call it, it should be during testing or something along these lines. We don't do testing during kernel loading.

Comment (xairy) [inline mm/ktsan/stack_depot.c]: We can only call this function only during early boot, since it uses memblock allocator, that's why it's marked __init. Otherwise the caller might expect that it can be called any time. I don't think we'll ever use it, so we can either remove it or leave it marked with __init as a reminder.

**Comment by dvyukov** [comment]
I think we just need to get of kt_stack_depot_destroy. It is not used and does not look useful during init.

**Comment by dvyukov** [comment]
Uploaded new version. PTAL

**Comment by xairy** [comment]
LGTM
